docs/tutorials/first-run-experience-tutorial.md

```diff
@@ -0,0 +1,152 @@
+
+    ```javascript
+    Office.onReady((info) => {
+      if (info.host === Office.HostType.Excel) {
+        document.getElementById("sideload-msg").style.display = "none";
+        document.getElementById("app-body").style.display = "flex";
+    
+        if (!localStorage.getItem("showedFRE")) {
+          showFirstRunExperience();
+        }
+    
+        document.getElementById("run").onclick = run;
+      }
+    });
+    
+    async function showFirstRunExperience() {
+      document.getElementById("first-run-experience").style.display = "flex";
+      localStorage.setItem("showedFRE", true);
+    }  
+    ```
+
+1. If you selected an Office application besides Excel, update the condition of the first `if` statement to check for your chosen [Office.HostType](/javascript/api/office/office.hosttype).
+
+### Update the CSS file
+
+Update the CSS file to ensure that the add-in UI is styled appropriately given the addition of the "first-run-experience" div.
+
+- Open the **taskpane.css** file. Replace the line `.ms-welcome__main {` with the following code, then save the file.
+
+    ```css
+    .ms-welcome__main, .ms-welcome__main > div {
+    ```
+
+### Try it out
+
+1. Ensure that the web server is running and the add-in has been sideloaded. For details, see the instructions in the quick start you used.
+
+1. If the add-in task pane isn't already open in Office, go to the **Home** tab and choose the **Show Taskpane** button on the ribbon to open it.
+
+1. Verify that the task pane includes the list of features.
+
+    ![The add-in task pane UI on first run.](../images/fre-tutorial-addin-first-run.png)
+
+1. Close the task pane then reopen it. Verify that the task pane no longer displays the list of features.
+
+    ![The add-in task pane UI on second run.](../images/fre-tutorial-addin-next-run.png)
+
+1. If you used `npm start` to start the local web server, run the following command when you want to stop the server and uninstall the add-in.
+
+    ```command&nbsp;line
+    npm stop
+    ```
+
+## Next steps
+
+Congratulations, you've successfully created an Office task pane add-in with a first-run experience!
+
+### Make it production ready
+
+Using this tutorial, you implemented a basic [first-run experience](../design/first-run-experience-patterns.md). For the first-run experience to be ready for users, you should consider the following:
+
+- Use a more secure and robust option for tracking first-run state. For example, use storage partitioning if available, or implement a Single Sign-on (SSO) authentication solution. For more about available settings options, see [Persist add-in state and settings](../develop/persisting-add-in-state-and-settings.md). For more about available authentication options, see [Overview of authentication and authorization](../develop/overview-authn-authz.md).
```

Comment:
Consider moving this bullet to the last one, since it's the least actionable in most cases.